tests/integration/cocoon/test_cocoon_holdings.py

```diff
@@ -680,9 +680,170 @@ def test_load_from_data_frame_holdings_success(
 
         self.assertEqual(len(responses["holdings"]["errors"]), 0)
 
+        # Assert that by default no unmatched_identifiers are returned in the response
+        self.assertFalse(responses["holdings"].get("unmatched_identifiers", False))
+
         self.assertTrue(
             expr=all(
-                isinstance(succcess_response.version, expected_outcome)
-                for succcess_response in responses["holdings"]["success"]
+                isinstance(success_response.version, expected_outcome)
+                for success_response in responses["holdings"]["success"]
             )
         )
+
+    @parameterized.expand(
+        [
+            [
+                "Standard successful load with no unmatched instruments",
+                "data/holdings-example-unique-date.csv",
+                True,
+                [],
+            ],
+            [
+                "Standard successful load with one unmatched instrument",
+                "data/holdings-example-unique-date-one-unmatched-instrument.csv",
+                True,
+                [
+                    {
+                        "Instrument/default/Sedol": "BFAKE712",
+                        "Instrument/default/Isin": "USFAHHDAR1014",
```

Comment:
is this a fake ISIN too? it should only return a LUID_ZZZZZZ if none of the identifiers match

you might actually need a test that covers this e.g. valid ISIN, invalid SEDOL should not return ZZZ and unmatched should be empty

Reply:
Good point. Test added, although the code handles this correctly already.